### PR TITLE
Hstar calculation move to confinement time functions.

### LINKED ISSUE #3534: `hstar` not properly calculated for every confinement scaling
This is a follow on from #3493.

The variable `hstar` which is meant to represent the $H$-factor value if the loss power has no radiation correction (`i_rad_loss = 3`) is only true if the IPB98(y,2) scaling is used.

The current implementation is:

```python
if physics_variables.i_rad_loss == 1:
            po.ovarrf(
                self.outfile,
                "H* non-radiation corrected",
                "(hstar)",
                physics_variables.hfact
                * (
                    physics_variables.p_plasma_loss_mw
                    / (
                        physics_variables.p_plasma_loss_mw
                        + physics_variables.pden_plasma_sync_mw
                        + physics_variables.p_plasma_inner_rad_mw
                    )
                )
                ** 0.31,
                "OP",
            )
        elif physics_variables.i_rad_loss == 0:
            po.ovarrf(
                self.outfile,
                "H* non-radiation corrected",
                "(hstar)",
                physics_variables.hfact
                * (
                    physics_variables.p_plasma_loss_mw
                    / (
                        physics_variables.p_plasma_loss_mw
                        + physics_variables.pden_plasma_rad_mw
                        * physics_variables.vol_plasma
                    )
                )
                ** 0.31,
                "OP",
            )
        elif physics_variables.i_rad_loss == 2:
            po.ovarrf(
                self.outfile,
                "H* non-radiation corrected",
                "(hstar)",
                physics_variables.hfact,
                "OP",
            )
```
- The calculation is also done in `outplas()` which is not recommended as `outplas()` should only be for output and not a place for doing calculations.
 - This also means that `hstar` is not a globally tracked physics variable which can be used in other parts of the code.

### PR DESCRIPTION
**Pull request by ajpearcey**
## Description

Closes #3534. I have moved Hstar calculation out of `outplas()`. I propose keeping the assumption that Hstar is computed with the IPB-98y2 scaling, as the way we currently calculated the energy confinement scaling means while possible there isn't an elegant solution. 

## Checklist

I confirm that I have completed the following checks:

- [ ] My changes follow the [PROCESS style guide](https://ukaea.github.io/PROCESS/development/standards/)
- [ ] I have justified any large differences in the regression tests caused by this pull request in the comments.
- [ ] I have added new tests where appropriate for the changes I have made.
- [ ] If I have had to change any existing unit or integration tests, I have justified this change in the pull request comments.
- [ ] If I have made documentation changes, I have checked they render correctly.
- [ ] I have added documentation for my change, if appropriate.

### CODE REVIEW

**Review by chris-ashe — APPROVED** [review]

**Review by timothy-nunn — CHANGES_REQUESTED** [review]
The quality jobs are failing
